lib/logstash/inputs/s3.rb

```diff
@@ -68,6 +70,15 @@ class LogStash::Inputs::S3 < LogStash::Inputs::Base
   # default to the current OS temporary directory in linux /tmp/logstash
   config :temporary_directory, :validate => :string, :default => File.join(Dir.tmpdir, "logstash")
 
+  # Specify a custom endpoint for use with non-AWS S3 implementations, e.g.,
+  # Ceph.  Provide a URL in the format http://127.0.0.1:8080/
+  config :endpoint, :validate => :string
```

Comment:
This will be available [through the mixin all the aws like plugins use](https://github.com/logstash-plugins/logstash-mixin-aws/blob/master/lib/logstash/plugin_mixins/aws_config/generic.rb#L29-L30)

So no need to implement it here.

Only the `force_path_style` setting below is s3 specific, can you respin this PR to only add that setting?